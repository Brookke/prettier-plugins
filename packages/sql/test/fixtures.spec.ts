import fs from 'fs'
import path from 'path'

import prettier from 'prettier'
import SqlPlugin from 'prettier-plugin-sql'

describe('parser and printer', () => {
  it('should format all fixtures', () => {
    const fixtures = path.resolve(__dirname, 'fixtures')
    for (const filepath of fs.readdirSync(fixtures)) {
      const input = fs.readFileSync(path.resolve(fixtures, filepath)).toString()

      const output = prettier.format(input, {
        filepath,
        parser: 'sql',
        plugins: [SqlPlugin],
      })

      expect(output).toMatchSnapshot(filepath)
    }
  })
})
