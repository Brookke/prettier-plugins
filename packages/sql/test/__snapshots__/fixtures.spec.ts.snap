// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser and printer should format all fixtures: basic.sql 1`] = `
"-- this is a comment
SELECT
  *
FROM
  tbl
WHERE
  foo = 'bar';

UPDATE
  a
SET
  id = 1
WHERE
  name IN (
    SELECT
      name
    FROM
      b
  )"
`;
